app/oracle.py

```diff
@@ -316,11 +354,46 @@ def run_once():
         logging.info('The tx hasn\'t been actually sent to the oracle contract! We are in DRY RUN mode')
         logging.info('Provide MEMBER_PRIV_KEY to be able to transact')
 
-    if not run_as_daemon:
-        logging.info('We are in single-iteration mode, so exiting. Set DAEMON=1 env to run in the loop.')
-        raise StopIteration()
 
-    logging.info(f'We are in DAEMON mode. Sleep {SLEEP} s and continue')
+def update_stable_swap_state_oracle_data():
+    logging.info('Check stable swap oracle state')
+    try:
+        oracle_price = stable_swap_state_oracle.functions.stethPrice().call()
+        pool_price = stable_swap_pool.functions.get_dy(1, 0, 10 ** 18).call()
+        percentage_diff = 100 * abs(1 - oracle_price / pool_price)
+        logging.info(
+            f'StETH stats: (pool price - {pool_price / 1e18:.6f}, oracle price - {oracle_price / 1e18:.6f}, difference - {percentage_diff:.2f}%)'
+        )
+
+        # Returns value in basis points: 10000 BP equal to 100%, 100 BP to 1%.
+        price_update_threshold = stable_swap_state_oracle.functions.priceUpdateThreshold().call() / 100
```

Comment:
I'd suggest calling `getProofParams()` here and get the price update threshold from there (the last return arg) instead. This would save one RPC call in the case when we need to update the oracle, and would still require only one call in the case when the oracle price is up-to-date.